### PR TITLE
Add direct tabular input support

### PR DESCRIPTION
**Pull request by vinhphatfsg**
## Summary
- Add CSV and Excel support to CLI direct mode, including headers, Excel range/sheet options, and format detection
- Add multi-field direct output via -F/--field and --output-map, with context support and resource guards
- Update README, rule specs, design doc, and CLI integration coverage

## Verification
- cargo fmt --check
- cargo test -p rulemorph_cli --test cli direct_
- cargo test
- codex review --uncommitted

## Security review
- Codex Security review initially found scanner edge cases around evaluated refs and CSV inference; fixes and regression tests are included
- Final Codex Security review found no blocking issues

### CODE REVIEW

File: crates/rulemorph_cli/src/direct/output_spec.rs [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+        }
+    }
+    for (index, field) in fields.iter().enumerate() {
+        if fields.iter().enumerate().any(|(other_index, other)| {
+            index != other_index
+                && (is_path_prefix(&field.tokens, &other.tokens)
+                    || is_path_prefix(&other.tokens, &field.tokens))
+        }) {
+            return Err("direct output target conflicts with another target".to_string());
+        }
```
[the remark this answers is not in the document]
Comment (vinhphatfsg) [reply crates/rulemorph_cli/src/direct/output_spec.rs]: 対応しました。親子 target 衝突検出をネストした全件比較から、path をソートして隣接 path だけを prefix 判定する実装に変更しています。先に large parent/child conflict の再現テストを追加し、旧実装で遅くなる事象を RED として確認してから修正しました。修正後は cargo fmt / 対象テスト / direct CLI integration tests / cargo test / Codex Security 差分レビューまで通しています。

File: crates/rulemorph_cli/src/direct.rs [hunk 2]
```diff
@@ -80,53 +130,415 @@ pub(crate) fn run(args: DirectArgs) -> i32 {
...
+    let core_range = format!(
+        "{}{}:{}{}",
+        data_range.start_col, header_row, data_range.end_col, data_range.end_row
+    );
+    let mut excel = serde_json::Map::new();
+    excel.insert("has_header".to_string(), serde_json::Value::Bool(true));
+    excel.insert(
+        "header_row".to_string(),
+        serde_json::Value::Number(header_row.into()),
+    );
```
[the remark this answers is not in the document]
Comment (vinhphatfsg) [reply crates/rulemorph_cli/src/direct.rs:442]: 確認しましたが、この指摘は再現しませんでした。`header_row.into()` 由来の compile error は発生しておらず、対象テストと workspace 全体の `cargo test` が通っています。そのためコード変更は不要と判断します。
